samples/dashboards/demos/custom-component/demo.js

```diff
@@ -0,0 +1,67 @@
+const Component = Dashboards.Component;
+
+class YouTubeComponent extends Component {
+    constructor(options) {
+        super(options);
+        this.type = 'YouTube';
+        this.youTubeElement = document.createElement('iframe');
+        return this;
+    }
+
+    resize(width, height) {
+        super.resize.call(this, width, height);
+        this.youTubeElement.setAttribute('width', width - 10);
+        this.youTubeElement.setAttribute('height', height - 10);
+    }
+
+    load() {
+        super.load();
+
+        this.youTubeElement.setAttribute(
+            'src',
+            'https://www.youtube.com/embed/' + this.options.videoId
+        );
+        this.youTubeElement.setAttribute('title', 'YouTube video player');
+        this.youTubeElement.setAttribute('frameborder', '0');
+        this.youTubeElement.allowfullscreen = true;
```

Comment:
In this part we mix two solutions for applying params, so I would suggest to use the `setAttribute` here, too.

Reply:
When I do it with `setAttribute` the error is thrown. Apart from that, that is only a demo :)